exonum/src/api/node/public/explorer.rs

```diff
@@ -276,6 +316,80 @@ impl ExplorerApi {
             })
     }
 
+    /// Return call status of committed transaction.
+    pub fn call_status(
+        snapshot: &dyn Snapshot,
+        query: CallStatusQuery,
+    ) -> Result<CallStatusResponse, ApiError> {
+        let explorer = BlockchainExplorer::new(snapshot);
+
+        let tx_info = explorer.transaction(&query.hash).ok_or_else(|| {
+            let description = serde_json::to_string(&json!({ "type": "unknown" })).unwrap();
+            ApiError::NotFound(description)
+        })?;
+        let tx_info = match tx_info {
+            TransactionInfo::Committed(info) => Ok(info),
+            TransactionInfo::InPool { .. } => {
+                let description =
+                    serde_json::to_string(&json!({ "type": "not committed" })).unwrap();
+                Err(ApiError::NotFound(description))
+            }
+        }?;
+
+        let block_height = tx_info.location().block_height();
+        let block_info = explorer.block(block_height).ok_or_else(|| {
+            ApiError::NotFound(format!(
```

Comment:
Why in some places error messages are encoded as JSON, and in some it's just a plain text?